### PR TITLE
enhance(ui): Stop content shift when scrollbars appear/disappear

### PR DESCRIPTION
**Pull request by zolrath**
Overlay Scrollbars were released into the wild effectively on accident, removed, readded and seem to be generally confused.

They have recently been [readded to Chrome](https://chromium-review.googlesource.com/c/chromium/src/+/2739606) and by using `overflow-y: overlay` we remove the content shift that occurs when the scrollbar appears and disappears.

In the future this problem can be solved using [scrollbar-gutter](https://developer.mozilla.org/en-US/docs/Web/CSS/scrollbar-gutter) but this feature is not yet implemented.

This video shows the shift before/after this change:

https://user-images.githubusercontent.com/454563/119237988-b36d3f80-baf4-11eb-86ec-83f25f2debbe.mp4

### CODE REVIEW

**Comment by zolrath** [comment]
I'd like to do a test of this with the web version of Athens as Firefox doesn't support `overlay` and I'd like to ensure proper fallbacks are in place.

**Comment by shanberg** [comment]
We can probably do this by getting the browser from the useragent at the top level and applying this to Firefox only, but we need a solution to Stylefy's parent selector problem first: https://github.com/Jarzka/stylefy/issues/59#issuecomment-846449821

**Comment by zolrath** [comment]
I think I've found a fairly simple solution in stylefy by using:
```clojure
   :overflow-y "auto"
   ::stylefy/supports {"overflow-y: overlay"
                      {:overflow-y "overlay"}}
```
I've maintained the previous behavior on the web version using Firefox while allowing Electron for Desktop and the web version using Chrome to benefit from the lack of content shift.

**Comment by tangjeff0** [comment]
Pleasee review @shanberg

File: src/cljs/athens/main/core.cljs [hunk 1]
```diff
@@ -40,6 +40,7 @@
                                                    :nodeIntegration true
                                                    :worldSafeExecuteJavaScript true
                                                    :enableRemoteModule true
+                                                   :enableBlinkFeatures 'OverlayScrollbars'
```
Comment (shanberg) [inline src/cljs/athens/main/core.cljs:43]: Could you add a comment here along the lines of "Remove when `scrollbar-gutter` is supported"?

**Review by shanberg — CHANGES_REQUESTED** [review]
Looks great, 2 qs

1. The build/style check should succeed if you run `cljstyle fix` locally
2. Do you have a clear sense of when to use `overlay` and when not to? There are a handful of other scroll areas in the app and I'm wondering if there's any reason not to update those too.

**Comment by zolrath** [comment]
> 1. The build/style check should succeed if you run cljstyle fix locally
Will do!

> 2. Do you have a clear sense of when to use `overlay` and when not to? There are a handful of other scroll areas in the app and I'm wondering if there's any reason not to update those too.

I went with the minimal amount of changes to fix the document shift but effectively overlay floats the scrollbar instead of making it take up space in the container.
The only reason I could see to not use it would be in situations where there is important content very close to the right edge of the content which would be obscured/look odd if the thin scrollbar was overlaid on top of it.

File: src/cljs/athens/main/core.cljs (hunk 1 above)
Comment (zolrath) [reply src/cljs/athens/main/core.cljs:45]: Done!

**Review by shanberg — APPROVED** [review]
Beautiful. Excellent little improvement.

**Comment by zolrath** [comment]
Is there a cljstyle configuration file? Running `cljstyle fix` recommends changes to comments in dozens of files, mostly related to changing some `;` comments to `;;` and some `;;;` comments to `;;`

**Review by tangjeff0 — APPROVED** [review]
Awesome first PR @zolrath !

**Pull request merged**